### PR TITLE
feat: add include_external_repositories and exclude_prefixes attributes to copy_to_directory

### PR DESCRIPTION
**Pull request by gregmagolan**
Also change the default behaviour such that files from external repositories are not included unless the external repository is listed in include_external_repositories

### CODE REVIEW

File: lib/private/copy_to_directory.bzl [hunk 1]
```diff
@@ -33,6 +33,25 @@ If there are multiple root paths that match, the longest match wins.
...
 target's package and and sub-packages are relative to the target's package and
 files outside of that retain their full workspace relative paths.
+""",
+    ),
+    "include_external_repositories": attr.string_list(
+        default = [],
+        doc = """
+List of external repository names to include in the output directory.
+
+By default no files from external repositories are copied into the output directory.
```
Comment (alexeagle) [inline lib/private/copy_to_directory.bzl:43]: can you give a little example here of an input file and how it gets copied?

File: lib/private/copy_to_directory.bzl [hunk 2]
```diff
@@ -33,6 +33,25 @@ If there are multiple root paths that match, the longest match wins.
...
+    ),
+    "exclude_prefixes": attr.string_list(
+        default = [],
+        doc = """
+List of path prefixes to exclude from output directory.
+
+If the output directory path for a file or directory starts with or is equal to
+a path in the list then that file is not copied to the output directory.
+
+Exclude prefixes are matched *before* and replace_prefixes are applied.
```
Comment (alexeagle) [inline lib/private/copy_to_directory.bzl:54]: dangling "and"

**Review by alexeagle — APPROVED** [review]

File: lib/private/copy_to_directory.bzl (hunk 2 above)
Comment (kormide) [inline lib/private/copy_to_directory.bzl:54]: Typo: `and`

**Comment by kormide** [comment]
LGTM. I'll let Alex approve once the example is in.

File: lib/private/copy_to_directory.bzl (hunk 2 above)
Comment (gregmagolan) [reply lib/private/copy_to_directory.bzl]: fixed

Comment (gregmagolan) [reply lib/private/copy_to_directory.bzl]: resolved

File: lib/private/copy_to_directory.bzl (hunk 1 above)
Comment (gregmagolan) [reply lib/private/copy_to_directory.bzl]: done

**Pull request merged**